analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/CollectionTracker.cs

```diff
@@ -146,4 +146,67 @@ public static ProgramState LearnFrom(ProgramState state, IPropertyReferenceOpera
         }
         return state;
     }
+
+    public static ProgramState LearnFrom(ProgramState state, IInvocationOperationWrapper invocation)
+    {
+        if (EnumerableCountConstraint(state, invocation) is { } constraint)
+        {
+            return state.SetOperationConstraint(invocation, constraint);
+        }
+        var targetMethod = invocation.TargetMethod;
+        if (invocation.Instance is { } instance && instance.Type.DerivesOrImplementsAny(CollectionTypes))
+        {
```

Comment:
```suggestion
        else if (invocation.Instance is { } instance && instance.Type.DerivesOrImplementsAny(CollectionTypes))
        {
            var targetMethod = invocation.TargetMethod;
```

Reply:
I moved the `targetMethod` inside the if.
About the `if` -> `else if`, there is a really funky issue:

the local function `SetOperationAndSymbolValue` needs `instance`. 
If you convert the `if` to `else if`, `instance` is not always available, so the code doesn't compile.
Try it locally, it will (kinda) make sense.
If you really want me to make it `if-else-if` instead of `if-return-if`, I will need to pass `instance`, which will make the 3 call sites of `SetOperationAndSymbolValue` a bit more verbose.

Personally I'm fine with both, let me know what you prefer.

Reply:
wild. Keep it as is.